index.js

```diff
@@ -422,6 +424,76 @@ module.exports = function math_plugin(md, options) {
         alt: ['paragraph', 'reference', 'blockquote', 'list']
     });
 
+    // Regex to capture any html prior to math block, the math block (single or multi line), and any html after the math block
+    const math_block_within_html_regex = /(?<html_before_math>[\s\S]*?)\$\$(?<math>[\s\S]+?)\$\$(?<html_after_math>(?:(?!\$\$[\s\S]+?\$\$)[\s\S])*)/gm;
+
+    // Regex to capture any html prior to math inline, the math inline (single line), and any html after the math inline
+    const math_inline_within_html_regex = /(?<html_before_math>[\s\S]*?)\$(?<math>.*?)\$(?<html_after_math>(?:(?!\$.*?\$)[\s\S])*)/gm;
+
+    md.core.ruler.push("math_block_in_html_block", (state) => {  
+        if (!enableMathBlockInHtml) {    
+            return false;
+        }
+        return handleMathInHtml(state, "math_block", "$$", math_block_within_html_regex);
+    });
+
+    md.core.ruler.after("math_block_in_html_block", "math_inline_in_html_block", (state) => {     
+        if (!enableMathInlineInHtml) {    
+            return false;
+        }
+        return handleMathInHtml(state, "math_inline", "$", math_inline_within_html_regex);
+    });
+
+    // For any html block that contains math, replace the html block token with new tokens that separate out
+    // the html blocks from the math
+    const handleMathInHtml = (state, mathType, mathMarkup, mathRegex,) => {
+        const tokens = state.tokens;
+
+        for (let index = tokens.length - 1; index >= 0; index--) {
+            const currentToken = tokens[index];
+            const newTokens = [];
+
+            if (currentToken.type !== "html_block") {
+                continue;
+            }
+
+            const content = currentToken.content;
+
+            // Process for each math referenced within the html block
+            for (const match of content.matchAll(mathRegex)) {
+                const html_before_math = match.groups.html_before_math;
+                const math = match.groups.math;
+                const html_after_math = match.groups.html_after_math;
+
+                if (html_before_math) {
+                    newTokens.push({ ...currentToken, type: "html_block", map: null, content: html_before_math });
```

Comment:
I think we need the `map` property for VS Code's markdown preview scroll sync. Can you compute this from the current token?

Reply:
I tried this out in VS Code as-is without any modifications to map and currently I observe that the markdown code editor and the markdown preview have scroll in sync for cases when math is used within html content of a markdown file and markdown cell.